dali/test/tensor_list_wrapper.h

```diff
@@ -35,19 +36,41 @@ class TensorListWrapper {
 
   template<typename Backend>
   const TensorList<Backend> *get() const {
-    DALI_ENFORCE(false,
-                 "Backend type not supported. You may want to write your own specialization");
+    DALI_FAIL("Backend type not supported. You may want to write your own specialization");
   }
 
+
   constexpr bool has_cpu() const noexcept {
     return cpu_ != nullptr;
   }
 
+
   constexpr bool has_gpu() const noexcept {
     return gpu_ != nullptr;
   }
 
 
+  std::string backend() const noexcept {
+    if (cpu_) {
+      return "cpu";
+    } else if (gpu_) {
+      return "gpu";
+    } else {
+      DALI_FAIL("Unknown backend. If you are here, something went terribly wrong");
+    }
+  }
+
+
+  const TensorList<CPUBackend>& cpu() const noexcept {
+    return *cpu_;
```

Comment:
You have DALI_ENFORCE on `get`, but here not - and you can easily return a dereferenced `nullptr`.

Reply:
I still don't get why you assert when returning pointer (the user will be perfectly happy getting nullptr probably) vs not asserting when returning reference (user can easily dereference nullptr and return it). It seems inconsistent.